frontend/public/components/configmap-and-secret-data.jsx

```diff
@@ -1,15 +1,83 @@
+import * as _ from 'lodash-es';
+
 import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import { Heading } from './utils';
+
+export const ConfigMapData = ({data}) => {
+	const dl = [];
+	Object.keys(data || []).sort().forEach(k => {
+		dl.push(<dt key={`${k}-k`}>{k}</dt>);
+		dl.push(<dd key={`${k}-v`}><pre className="co-pre-wrap">{data[k]}</pre></dd>);
+	});
+
+	return <dl>{dl}</dl>;
+};
+
+export class SecretData extends React.PureComponent {
+	constructor(props) {
+		super(props);
+		this.state = { showSecret: false };
+		this.toggleSecret = this.toggleSecret.bind(this);
+	}
 
-const ConfigMapAndSecretData = ({data, decode}) => {
-  decode = decode || (v => v);
+	toggleSecret() {
+		this.setState({ showSecret: !this.state.showSecret })
+	}
 
-  const dl = [];
-  Object.keys(data || []).sort().forEach(k => {
-    dl.push(<dt key={`${k}-k`}>{k}</dt>);
-    dl.push(<dd key={`${k}-v`}><pre className="co-pre-wrap">{decode(data[k])}</pre></dd>);
-  });
+	render() {
+		const { data, type, decode } = this.props;
+		const { showSecret } = this.state;
+		const dl = [];
+		Object.keys(data).sort().forEach(k => {
+			if (k === '.dockercfg' || k === ".dockerconfigjson") {
```

Comment:
I'm hesitant to add back special support for this since it has caused bugs in the past for us. It also in a way obfuscates what's really in the secret. I'm not sure this is better than just showing the docker config json. Can you show a before and after screen?

Reply:
@spadgett so here is the old **dockercfg**
```
{  
   "https://index.docker.io/v1/":{  
      "username":"name",
      "password":"pass",
      "email":"mail@test",
      "auth":"bmFtZTpwYXNz"
   }
}
```

and here is the newer **dockerconfigjson**
```
{
	"auths": {
		"https://index.docker.io/v1/": {
			"auth": "dGVzdHVzZXI6dGVzdHB3",
			"email": "jhadvig@test.com"
		},
		"https://index.docker.io/v2": {
                        "auth": "dGVzdHVzZXI6dGVzdHB3",
                        "email": "test@test.com"
                }
	}
}
```

Reply:
before:
![before](https://user-images.githubusercontent.com/1668218/40144051-846a6d08-595d-11e8-803c-2877d7f9e000.png)

after:
![after](https://user-images.githubusercontent.com/1668218/40144060-8f4086b8-595d-11e8-87a1-76c0edc0c856.png)

Reply:
Per @kans comment, I think we should pull that change out as a separate PR to discuss. I'm not sure we shouldn't just display the raw JSON.

Reply:
yeah this is an opportunity for us to get rid of some weird usability cruft we built up over the years. Treating certain secrets as special creates an inconsistent experience and makes the code much harder to maintain